extensions/ql-vscode/src/pure/time.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Contains an assortment of helper functions for working with time, dates, and durations.
+ */
+
+
+const durationFormatter = new Intl.RelativeTimeFormat('en', {
+  numeric: 'auto',
+});
+
+// All these are approximate, specifically months and years
+const MINUTE_IN_MILLIS = 1000 * 60;
+const HOUR_IN_MILLIS = 60 * MINUTE_IN_MILLIS;
+const DAY_IN_MILLIS = 24 * HOUR_IN_MILLIS;
+const MONTH_IN_MILLIS = 30 * DAY_IN_MILLIS;
+const YEAR_IN_MILLIS = 365 * DAY_IN_MILLIS;
+
+export function humanizeDuration(diffInMs?: number) {
+  if (diffInMs === undefined) {
+    return '';
+  }
+
+  if (Math.abs(diffInMs) < HOUR_IN_MILLIS) {
+    return durationFormatter.format(Math.floor(diffInMs / MINUTE_IN_MILLIS), 'minute');
+  } else if (Math.abs(diffInMs) < DAY_IN_MILLIS) {
+    return durationFormatter.format(Math.floor(diffInMs / HOUR_IN_MILLIS), 'hour');
+  } else if (Math.abs(diffInMs) < MONTH_IN_MILLIS) {
+    return durationFormatter.format(Math.floor(diffInMs / DAY_IN_MILLIS), 'day');
+  } else if (Math.abs(diffInMs) < YEAR_IN_MILLIS) {
+    return durationFormatter.format(Math.floor(diffInMs / MONTH_IN_MILLIS), 'month');
+  } else {
+    return durationFormatter.format(Math.floor(diffInMs / YEAR_IN_MILLIS), 'year');
+  }
+}
+
+function createFormatter(unit: string) {
+  return Intl.NumberFormat('en-US', {
+    style: 'unit',
+    unit,
+    unitDisplay: 'long'
+  });
+}
+
+export function humanizeUnit(diffInMs?: number): string {
```

Comment:
I have a very tired brain, but I can't easily understand the difference between `humanizeDuration` and `humanizeUnit` - can we think of different names?

Reply:
I'll add comments. I can't think of better names.